src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -92,6 +92,7 @@ CreateShardConnectionHash(void)
 	info.keysize = sizeof(int64);
 	info.entrysize = sizeof(ShardConnections);
 	info.hash = tag_hash;
+	info.hcxt = CurrentMemoryContext;
```

Comment:
I'd be more comfortable with using TopTransactionContext. Do you see any arguments against that?